index.js

```diff
@@ -21,18 +24,45 @@ $('#storageModal').on('show.bs.modal', () => {
   showTileList();
 });
 
-const map = L.map('map');
+
+// RD Dutch projection
+let projection = L.CRS.EPSG3857;
+if (isTMS) {
+  projection = new L.Proj.CRS('EPSG:28992', '+proj=sterea +lat_0=52.15616055555555 +lon_0=5.38763888888889 +k=0.9999079 +x_0=155000 +y_0=463000 +ellps=bessel +units=m +towgs84=565.2369,50.0087,465.658,-0.406857330322398,0.350732676542563,-1.8703473836068,4.0812 +no_defs',
+    {
+      resolutions: [3440.640, 1720.320, 860.160, 430.080, 215.040, 107.520, 53.760, 26.880, 13.440, 6.720, 3.360, 1.680, 0.840, 0.420, 0.210],
+      bounds: L.bounds([-285401.92, 22598.08], [595401.9199999999, 903401.9199999999]),
+      origin: [-285401.92, 22598.08],
+    });
+}
+
+const map = L.map('map', {
+  crs: projection,
+});
+let baseLayer;
+
 // offline baselayer, will use offline source if available
-const baseLayer = L.tileLayer
-  .offline(urlTemplate, {
-    attribution: 'Map data {attribution.OpenStreetMap}',
-    subdomains: 'abc',
-    minZoom: 13,
-  })
-  .addTo(map);
+if (map.getContainer().classList.contains('tms')) {
```

Comment:
I think you could reuse isTMS here. This example was already complex for it's purpose, is it possible to split some logic in separate js files (one for the old, one for the tms page, and maybe one with shared code) ?